### ISSUE TITLE
How to improve the text detection model psenet

### ISSUE THREAD

**Issue by parulsingh23** [open-mmlab/mmocr#377]
After running the KIE sdmgr model on a receipt annotation provided by text detection model psenet, I have been getting some weird text outputs. For example, for a (clearly shown) printed item cost of '16.87', the psenet model outputs '16.6' and '.87'. 
I am confused as to how I can prevent such errors: is it that I'll need to create a new training dataset of receipt images for the psenet model, or is another model better suited to read receipts/to have its annotations input to the sdmgr model? Is my problem caused because I am using a model under 'text detection' (is it instead advisable to use a model under 'text recognition'- reason I haven't is that the 'text recognition' models seem to cater towards reading very distinct/logo like texts, and most receipts are just serif fonts)

So sorry for the barrage of question! These models have been super helpful, I'm just not sure I'm selecting the appropriate model for the job. I'd appreciate any suggestions!

**Comment by cuhk-hbsun** [comment]
Thanks for your feedback.
1. Could you please attach a single receipt image here?
2. I am not sure that there are other models which have better performance. You can have a try using other models.
3. It will be better to refine the model using your own data, using the released model as pretrained.

**Comment by parulsingh23** [comment]
Thank you for the reply! I realize that the text reading error follows from the text detection model (psenet) to the text recognition model (sar) (my understanding is that KIE is using those models, definitely correct me if I'm wrong). For whatever reason, the bboxes in this receipt separated the text of '16.' and '.87' (right around the number 8 & the decimal point), this is located right across from 'Boys Shoes':
![IMG_20210719_162518078](https://user-images.githubusercontent.com/55006754/126885515-bdba4c34-f703-4b27-9203-1d488bdf3d00.jpg)

I do understand that Walmart receipts are pretty standard and they were included in the training set (49 Walmart Receipts), which led me to assume the KIE model would be very accurate on such receipt extraction. Considering that I'm expect to use this model for some stores pretty commonly (ex. Walmart, Target, Costco, Grocery Stores), and I'd really like to get as high an accuracy as possible for these much-used store receipts, is a template better applied here? If so, would you have any pointers as to how I should integrate such a template in the KIE model?

Alternatively, if I am going to train the KIE model on these receipts, is there an expected number of say, Walmart receipts, that you would expect I would need, until the probability of getting the store name/address/phone number, items, quantities, prices, taxes, and totals _all_ correct on a single Walmart receipt is >85%? (I would just like to understand the approx. number of 'Walmart Receipts' I should train the model on)

**Comment by parulsingh23** [comment]
@cuhk-hbsun, After looking around the codebase, I believe the issue of misreading is primarily due to my text detection model (as if that choose incorrect bboxes as in my Walmart Receipt above, then the text recognition model will also generate wrong text, and the kie model won't be as good either). I believe using a text detection model trained on the ICDAR2017 dataset would be better (as that year's competition topic was around 'documentation' rather than just 'scenes'-- which would be more significant to me as the kie model looks at receipts). Please let me know if the idea of substituting the psenet with mask rcnn makes sense (as that is the only text detection model that was trained on ICDAR2017). Thanks!
